File: src/news_briefing/delivery/deeplinks.py
"""증권사 앱 딥링크 생성 (F19).

국내 종목(KRX 6자리 stock_code)만 지원. 해외는 미지원.
주문 실행까지 이동하는 편의 기능일 뿐, 체결은 사용자 손 (CLAUDE.md P1).
"""
from __future__ import annotations


def build_deeplinks(stock_code: str) -> dict[str, str]:
    if not stock_code:
        return {}
    return {
        "toss": f"supertoss://stock/{stock_code}",
        "koreainvestment": f"koreainvestment://stock/{stock_code}",
        "naver": f"https://m.stock.naver.com/domestic/stock/{stock_code}/total",
    }

File: frontend/src/lib/deeplinks.ts
export interface DeeplinkSet {
  toss: string
  koreainvestment: string
  naver: string
}

export interface TickerLinkSet {
  label: string
  href: string
}

export function buildDeeplinks(stockCode: string): DeeplinkSet | null {
  if (!stockCode) return null
  return {
    toss: `supertoss://stock/${stockCode}`,
    koreainvestment: `koreainvestment://stock/${stockCode}`,
    naver: `https://m.stock.naver.com/domestic/stock/${stockCode}/total`,
  }
}

/** raw 티커 → 증권사·정보 링크 목록 (국내/해외 자동 판별) */
export function buildTickerLinks(ticker: string): TickerLinkSet[] {
  if (!ticker) return []
  const isDomestic = /^\d{6}$/.test(ticker)
  if (isDomestic) {
    return [
      { label: '토스증권', href: `supertoss://stock/${ticker}` },
      { label: '네이버증권', href: `https://m.stock.naver.com/domestic/stock/${ticker}/total` },
    ]
  }
  // 지수·환율·선물은 링크 불필요
  if (/^\^/.test(ticker) || /=X$/.test(ticker) || /=F$/.test(ticker)) return []
  // 미국 주식·ETF
  return [
    { label: '네이버증권', href: `https://m.stock.naver.com/worldstock/stock/${ticker}/total` },
    { label: 'Yahoo Finance', href: `https://finance.yahoo.com/quote/${ticker}` },
  ]
}
